src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -70,16 +70,77 @@ object RegressionUtils {
     (y, cov, completeSamples)
   }
 
-  def buildGtColumn(gts: Iterable[Option[Int]]): Option[DenseMatrix[Double]] = {
-    val (nCalled, gtSum, allHet) = gts.flatten.foldLeft((0, 0, true))((acc, gt) => (acc._1 + 1, acc._2 + gt, acc._3 && (gt == 1) ))
+  def mutateLastColumn(X: DenseMatrix[Double], gts: IntIterator, mask: Array[Boolean]): Boolean = {
+    require(X.offset==0 && X.majorStride==X.rows && !X.isTranspose)
 
-    // allHomRef || allHet || allHomVar || allNoCall
-    if (gtSum == 0 || allHet || gtSum == 2 * nCalled || nCalled == 0 )
+    val n = X.rows
+    val k = X.cols - 1
+    var missingIndices = new mutable.ArrayBuilder.ofInt()
+    var i = 0
+    var j = k * n
+    var gtSum = 0
+    while (i < mask.length) {
+      val gt = gts.nextInt()
+      if (mask(i) && gt != -1) {
+        gtSum += gt
+        X.data(j) = gt.toDouble
+        j += 1
```

Comment:
No, because there is a third case (mask(i)==false) when we don't add to j.

I structured the conditional as I did because mask(i) && gt != -1 should be by far the most common situation.  Alternatively I could also nest conditionals:

```
    if (mask(i) {
      if (gt != -1) {
        gtSum += gt
        X.data(j) = gt.toDouble
      } else
        missingIndices += j
      j += 1
    }
```